src/js.cookie.js

```diff
@@ -88,7 +88,14 @@
 			// in case there are no cookies at all. Also prevents odd result when
 			// calling "get()"
 			var cookies = document.cookie ? document.cookie.split('; ') : [];
-			var rdecode = /(%[0-9A-Z]{2})+/g;
+
+			// This regex matches all valid values for UTF-8 percent encoding.  See https://tools.ietf.org/html/rfc3629
+			// Values U+0000 - U+007F are 1 byte encoded
+			// Values U+0080 - U+07FF are 2 byte encoded.  The first byte begins with 110 ([CD]).  The continuation byte begins with 10 ([89AB])
+			// Values U+8000 - U+FFFF are 3 byte encoded.  The first byte begins with 1110 (E).  The continuation byte begins with 10 ([89AB])
+			// Values U+10000 - U+FFFFF are 4 byte encoded.  The first byte begins with 111100 (F[0-3]).  The continuation byte begins with 10 ([89AB])
+			// Values U+100000 - U+10FFFF are 4 byte encoded.  The first byte is 11110100 (F4).  The second byte begins with 1000 (8).  The continuation byte begins with 10 ([89AB])
+			var rdecode = /(%[0-7][0-9A-F])|(%[CD][0-9A-F]%[89AB][0-9A-F])|(%E[0-9A-F]%[89AB][0-9A-F]%[89AB][0-9A-F])|(%F3%[89AB][0-9A-F]%[89AB][0-9A-F]%[89AB][0-9A-F])|(%F4%8[0-9A-F]%[89AB][0-9A-F]%[89AB][0-9A-F])+/gi;
```

Comment:
> The cookie's being created by an ASP.NET MVC web project, using a standard method (UrlEncode) https://msdn.microsoft.com/en-us/library/system.web.httputility.urlencode(v=vs.110).aspx. Why they chose to lowercase the hex values is unclear, but as you noted above it is spec compliant.

I understand. The problem is that the amount of unminifiable content that is added in the code will make it bigger than 900 bytes gzipped, so I am definitely not willing to add such amount of code if the problem can be fixed using converters. Out of the box, js-cookie supports only what it produces, maybe we can update the docs to make that clear.

I don't use ASP.NET, therefore, can you add the peculiarities of ASP.NET in the [SERVER_SIDE](https://github.com/js-cookie/js-cookie/blob/f2b25713e8fae658d418b60710adc0e0d14998d9/SERVER_SIDE.md) documentation along with the proper converter to make it handle ASP.NET cookie, just like what was done with the PHP one? That would be much appreciated!